### ISSUE TITLE
Run as daemon

### ISSUE THREAD

**Issue by radutopala** [ochinchina/supervisord#16]
Is there a way to run as a daemon?

Seems like the child processes do not see the exit signal from the supervisor parent. If I kill the supervisor via `kill -9`, the child processes continue to exist.

**Comment by ochinchina** [comment]
I make a minor change for killing all children if the supervisor exit.

Please check if it works in your environment.

**Comment by radutopala** [comment]
@ochinchina see https://github.com/ochinchina/supervisord/commit/5a470cc662fc08aa4066db078a6c5a02802d929f#commitcomment-21909793

**Comment by radutopala** [comment]
Solution: https://github.com/kubernetes/kubernetes/pull/13596/files

**Comment by radutopala** [comment]
@ochinchina still not closing all children if I kill the supervisor pid.

**Comment by ochinchina** [comment]
in the Linux environment, the children will be closed if the supervisor is killed but I don't know how to do this in OSX.

Do you have any idea on the OSX?

**Comment by ochinchina** [comment]
is possible for you send other signals other than SIGKILL to supervisor to let it gracefully shutdown?

**Comment by yokujin** [comment]
Why kill daemon by SIGKILL? Isn't it too rude?

Daemon mode was implemented in #31. Also added some client commands to gracefully shutdown daemon.
I think it's better to stop daemon via `supervisord ctl shutdown` command.

**Comment by radutopala** [comment]
I'm on OSX. Just updated to latest, ran `supervisord -d -c ../config.file`. Then I ran `supervisord ctl shutdown`, the services are still up..

**Comment by yokujin** [comment]
Had `supervisord` itself gone down?

**Comment by radutopala** [comment]
Nope..

**Comment by yokujin** [comment]
Does `supervisord ctl stop <process_name>` work?

**Comment by radutopala** [comment]
```
supervisord -c=config/supervisor/dev.conf ctl stop worker_0
worker_0: failed [Post http://localhost:9001/RPC2: dial tcp [::1]:9001: getsockopt: connection refused]
```
Hmm, I think I'm missing something.. should I do anything specific?

**Comment by yokujin** [comment]
Do following lines exist in config?
```
[inet_http_server]
port=127.0.0.1:9001
```

**Comment by radutopala** [comment]
@yokujin thanks, that did the job, now `supervisord ctl shutdown` did stop everything. We should document all the commands in README at least.

**Comment by yokujin** [comment]
Probably we'll do. 
I don't know if it's really necessary as all documentation already published on original python supervisord daemon's [site](http://supervisord.org/)

**Comment by radutopala** [comment]
@yokujin well, for example, the `ctl` is implemented as a separate binary on the Py version http://supervisord.org/running.html#running-supervisorctl. For newcomers here, on the Go version, would be helpful to know about the `supervisord ctl` and the necessity of having the RPC turned on. Should I push a MR with this doc?

**Comment by radutopala** [comment]
@yokujin also, could the daemon be used with init systems?

**Comment by yokujin** [comment]
Yeah, `ctl` subcommand is documented nowhere. It is only mentioned in `supervisord --help` output.

I do have a plan to write documentation on all implemented commands. But I'm slightly short of time at the moment.
If you could do it - you should do it. :smile:

**Comment by yokujin** [comment]
I use it on CentOS 6 with theirs initscripts. 
The only thing I had to do is to run it with specific user directly from init script.
Example init script:
```
#!/bin/bash
#
# proggy_name   This scripts turns proggy_name on
#
# Author:       Proud User <email@anywhere.com> (based off supervisord)
#
# chkconfig:	- 95 04
#
# description:  supervisor is a process control utility.  It has a web based
#               xmlrpc interface as well as a few other nifty features.
# processname:  proggy_name
#

# source function library
. /etc/rc.d/init.d/functions

prog="proggy_name"

[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog

config=$TOPDIR/supervisord.conf
pidfile=$TOPDIR/log/supervisord.pid

RETVAL=0

start() {
	echo -n $"Starting proggy_name: "
	daemon --check proggy_name --user $RUN_AS_USER /path/to/supervisord --configuration $config --daemon
	RETVAL=$?
	echo
	[ $RETVAL -eq 0 ] && touch /var/lock/subsys/proggy_name
}

stop() {
	echo -n $"Stopping proggy_name: "
	/path/to/supervisord ctl shutdown > /dev/null 2>&1
	RETVAL=$?
	echo
	[ $RETVAL -eq 0 ] && rm -f /var/lock/subsys/proggy_name
}

restart() {
	stop
	start
}

case "$1" in
  start)
	start
	;;
  stop)
	stop
	;;
  restart|force-reload|reload)
	restart
	;;
  condrestart)
	[ -f /var/lock/subsys/proggy_name ] && restart
	;;
  status)
	status -p $pidfile proggy_name
	RETVAL=$?
	;;
  *)
	echo $"Usage: $0 {start|stop|status|restart|reload|force-reload|condrestart}"
	exit 1
esac

exit $RETVAL
```

**Comment by seanlook** [comment]
Why the child programs shoud exit when the supervisord exit?

I think the child managed programs shoud be kept alive, whatever the supervisord process state is.

**Comment by moooofly** [comment]
> Why the child programs shoud exit when the supervisord exit?
> 
> I think the child managed programs shoud be kept alive, whatever the supervisord process state is.

If children processes still be alive after supervisord exit, they will be out of control. Besides, when supervisord starts again, there will be conflics when starting new childrem processes.
